### DISCUSSION TITLE
Automatic laser detection for video

### DISCUSSION THREAD

**Discussion by Paradis-Hautcoeur** [biigle/core#721] [Q&A]
Hi, I know BIIGLE can detect laser on still images, I was wondering if some of you found a way to do it on video ? 

I am now manually tagging laser every ten seconds, I wish it would be at every five seconds but it would take too much time. I want to mesure the consistancy of the area filmed.

Thank you for your help !

**Comment by mzur** [comment]
Here is the BIIGLE perspective on this topic: In our experience, videos are mostly captured in a way that makes it impossible to use the laser information for area estimation. If the video is captured e.g. with an oblique angle to the ground you may be able to estimate the size of a flat object that is directly illuminated by the laser points but anything closer or farther away cannot be estimated. This is the reason why there is no laser point detection for videos in BIIGLE (although we [keep this issue in mind](https://github.com/biigle/laserpoints/issues/46)).

Of course you can try to do *something* with the laser points outside of BIIGLE if you don't care that the area estimation cannot be trusted. Maybe in your case you don't want the absolute area but only want to compare the "values" between the video frames. BIIGLE uses the (by now slightly aged) [DELPHI method](https://doi.org/10.3389/fmars.2015.00020) for laser point detection. This method can be implemented and used on individual video frames outside BIIGLE, too. If you have lots of annotated laser points as training data, there may be more recent approaches for general object detection that could provide better results (which we also want to [explore](https://github.com/biigle/laserpoints/issues/44) sometime).
